sympy/simplify/trigsimp.py

```diff
@@ -1098,19 +1098,19 @@ def futrig(e, **kwargs):
         return e
 
     old = e
-    e = bottom_up(e, lambda x: _futrig(x, **kwargs))
+    e = bottom_up(e, _futrig)
 
     if kwargs.pop('hyper', True) and e.has(HyperbolicFunction):
         e, f = hyper_as_trig(e)
-        e = f(_futrig(e))
+        e = f(bottom_up(e, _futrig))
```

Comment:
This results in a worse simplification when
This results in a worse simplification when
```
e = -t*sin(_Dummy*alpha/2)**2 + t*cos(_Dummy*alpha/2)**2 - 2*I*x*sin(_Dummy*alpha/2)*cos(_Dummy*alpha/2)
```
(`_Dummy` inserted by `hyper_as_trig`). The results are
```
ipdb> _futrig(e)
t*cos(_Dummy*alpha) - I*x*sin(_Dummy*alpha)
ipdb> bottom_up(e, _futrig)
-2*t*sin(_Dummy*alpha/2)**2 + t - I*x*sin(_Dummy*alpha)
```
See #19917 for a full repro